zig fmt fails to parse return values when missing parenthesis

Running `zig fmt` on:
```zig
const mycond = true;
pub fn foo() if (mycond) i32 else void {
    if (mycond) {
        return 42;
    }
}
```

fails with: `fmt-fail.zig:7:1: error: Expected ';' or '{', found Eof`

This works okay though:
```zig
const mycond = true;
pub fn foo() (if (mycond) i32 else void) {
    if (mycond) {
        return 42;
    }
}
```

--- a/std/zig/parse.zig
+++ b/std/zig/parse.zig
@@ -2833,8 +2833,8 @@ fn parseIf(arena: *Allocator, it: *TokenIterator, tree: *Tree, bodyParseFn: Node
 
     const else_token = eatToken(it, .Keyword_else) orelse return node;
     const payload = try parsePayload(arena, it, tree);
-    const else_expr = try expectNode(arena, it, tree, parseExpr, AstError{
-        .ExpectedExpr = AstError.ExpectedExpr{ .token = it.index },
+    const else_expr = try expectNode(arena, it, tree, bodyParseFn, AstError{
+        .InvalidToken = AstError.InvalidToken{ .token = it.index },
     });
     const else_node = try arena.create(Node.Else);
     else_node.* = Node.Else{

--- a/std/zig/parser_test.zig
+++ b/std/zig/parser_test.zig
@@ -2234,6 +2234,18 @@ test "zig fmt: multiline string in array" {
     );
 }
 
+test "zig fmt: if type expr" {
+    try testCanonical(
+        \\const mycond = true;
+        \\pub fn foo() if (mycond) i32 else void {
+        \\    if (mycond) {
+        \\        return 42;
+        \\    }
+        \\}
+        \\
+    );
+}
+
 const std = @import("std");
 const mem = std.mem;
 const warn = std.debug.warn;

